Equinor.Maintenance.API.EventEnhancer/EventSourcing/SourceReactor.cs

```diff
@@ -0,0 +1,55 @@
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using Azure;
+using Equinor.Maintenance.API.EventEnhancer.Constants;
+using Equinor.Maintenance.API.EventEnhancer.ExtensionMethods;
+using Equinor.Maintenance.API.EventEnhancer.MaintenanceApi.Clients;
+
+namespace Equinor.Maintenance.API.EventEnhancer.EventSourcing;
+
+public class SourceReactor(IHttpClientFactory factory)
+{
+    public async Task<(JsonObject, Uri)> Enhance(string eventType, string objectId)
+    {
+        var request = eventType switch
+        {
+            "BUS2007" => EnhanceWorkOrder(objectId),
+            "BUS2038" => EnhanceWorkOrder("notimplemented"),
+            _ => throw new ArgumentOutOfRangeException(nameof(eventType))
+        };
+        return await request;
+    }
+
+// rewrite this method to be more generic (work for any type?) and to also construct objects from an allowlist
+    private async Task<(JsonObject, Uri)> EnhanceWorkOrder(string objectId)
+    {
+        var workOrderClient   = new WorkOrderClient(factory.CreateClient(Names.MainteanceApi));
```

Comment:
Typo, use refactor functionality in Rider

```suggestion
        var workOrderClient   = new WorkOrderClient(factory.CreateClient(Names.MaintenanceApi));
```